src/components/CommentForm.jsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from "react";
+import { addComment } from "../api";
+
+export const CommentForm = ({ articleId, comments, setComments }) => {
+  const [usernameInput, setUsernameInput] = useState("");
+  const [commentInput, setCommentInput] = useState("");
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [error, setError] = useState(null);
+  const [usernameErrorMsg, setUsernameErrorMsg] = useState(null);
+  const [commentErrorMsg, setCommentErrorMsg] = useState(null);
+
+  const handleUsernameBlur = () => {
+    if (usernameInput.length < 2) {
+      setUsernameErrorMsg('Username must be longer than 1 character.');
+    } else {
+      setUsernameErrorMsg('');
+    }
+  };
+  
+  const handleCommentBlur = () => {
+    if (commentInput.length === 0) {
+      setCommentErrorMsg('Insert Comment');
+    } else {
+      setCommentErrorMsg('');
+    }
+  };
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const now = new Date();
+    const isoString = now.toISOString();
+    setError(null);
+    setIsSubmitted(true);
+    setComments((currentComments) => [
+      { author: usernameInput, body: commentInput, created_at: isoString },
+      ...currentComments,
+    ]);
+    addComment(articleId, usernameInput, commentInput).catch((err) => {
+      setIsSubmitted(false);
+      setError(err);
+      setComments(comments);
+    });
+    setUsernameInput('')
+    setCommentInput('')
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <h3>Add Comment</h3>
+      <label>
+        Username
+        <input
+          id="username-input"
+          value={usernameInput}
+          onChange={(event) => setUsernameInput(event.target.value)}
+          onBlur={handleUsernameBlur}
+        />
+      </label>
+      <p id="error-msg">{usernameErrorMsg}</p>
+      <label>
+        Comment
+        <input
+          id="comment-input"
+          value={commentInput}
+          onChange={(event) => setCommentInput(event.target.value)}
+          onBlur={handleCommentBlur}
+        />
+      </label>
+      <p id="error-msg">{commentErrorMsg}</p>
+      <button
+        type="submit"
+        disabled={commentInput.length < 1 || usernameInput.length < 1}
+      >
+        SUBMIT
+      </button>
+      {isSubmitted ? (
+        <div id="success-msg">
+          <p>Comment Submitted</p>
+        </div>
+      ) : null}
+      {error ? (
+        <div id="error-msg">
+          <p>Something went wrong, please try again.</p>
+          <p>Username may not be valid.</p>
+          <p>
+            {error.response.status} {error.response.data.msg}
```

Comment:
Currently, if I try and submit a comment using a non existent username, I'm seeing the `Comment submitted` message which then changes to the error. Ideally, a user shouldn't see the success message at all if the posting has failed. I would say you can achieve the posting of a comment without optimistic rendering in this and have the success or error display depending on the result of the post request. You can still provide feedback to the user when a comment is posting, like posting... or even disabling the submit button